core/src/prettier/mod.rs

```diff
@@ -0,0 +1,119 @@
+mod document;
+mod print;
+
+pub use self::document::{concat, const_text, display, flatten, indent, nl, text, Document};
```

Comment:
Question: why use `self::document` rather than just `document`?

Reply:
The `rust-analyzer`/`rustfmt` configuration that matches our conventions will insert dependencies with `crate/super/self` prefixes, even though the `rustfmt.toml` has that disabled currently. I have it enabled in my editor though, and I it find very helpful when skimming imports. It visually distinguishes local imports vs third-party crates, and ensures imports are always unambiguous (i.e. there can be cases where a dependency and an internal module have the same name). That can be handled on a case-by-case basis, but I don't see any real reason to change this. In any case though, once we enable that configuration in our `rustfmt` config, this is what imports will end up looking like anyway.